test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1407,21 +1408,16 @@
                                                :dns-alt-names dns-alt-names))))))
 
     (testing "basic extensions are created for a CA"
-      (let [serial        42
-            exts          (create-ca-extensions subject-dn
-                                                serial
+      (let [exts          (create-ca-extensions subject-pub
                                                 subject-pub)
             exts-expected [{:oid      "2.16.840.1.113730.1.13"
                             :critical false
                             :value    netscape-comment-value}
-                           {:oid       "2.5.29.17"
-                            :critical false
-                            :value    {:dns-name [subject]}}
                            {:oid      "2.5.29.35"
                             :critical false
-                            :value    {:issuer-dn     (str "CN=" subject)
-                                       :public-key    nil
-                                       :serial-number (biginteger serial)}}
+                            :value    {:issuer-dn     nil
+                                       :public-key    subject-pub
```

Comment:
Does the value of the AuthorityKeyIdentifier extension (with oid `2.5.29.35`) contain the raw public key or is it a value derived from the public key as described in https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1:

> The value of the keyIdentifier field SHOULD be derived from the public key used to verify the certificate's signature or a method that generates unique values, e.g. Two common methods for generating key identifiers from the public key are described in Section 4.2.1.2.

Where https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.2 describes type 1 and 2:

> Two common methods for generating key identifiers from the public key are:
> 
>       (1) The keyIdentifier is composed of the 160-bit SHA-1 hash of the
>            value of the BIT STRING subjectPublicKey (excluding the tag,
>            length, and number of unused bits).
> 
>        (2) The keyIdentifier is composed of a four-bit type field with
>            the value 0100 followed by the least significant 60 bits of
>            the SHA-1 hash of the value of the BIT STRING
>            subjectPublicKey (excluding the tag, length, and number of
>            unused bits).
>

Reply:
From what I can tell, the extension has the raw key, which is then processed into the SHA1 hash. See [this test](https://github.com/puppetlabs/jvm-ssl-utils/blob/4c47e1bb680c07aa54e90ad987e7bdc753ec2d73/test/puppetlabs/ssl_utils/core_test.clj#L312-L323). I can add a test in this repo to verify that if we want.

Reply:
As indicated in https://tickets.puppetlabs.com/browse/SERVER-2662, we use the first method to generate these IDs.

Reply:
ah cool, makes sense!

Reply:
I added a test for this.

Reply:
And this surprised me too when I first noticed it. I had written the first test originally checking for that SHA1, and it failed of course.